templates/add_user.html

```diff
@@ -39,23 +41,75 @@
   <br><br>
 
 
-  <form id="users-group-form" method="get"
-    action="{{ BASE_URL }}{{ user_add_path }}">
-    <paper-input label="Input a group key (group email address or unique id) to fetch more users." type="textbox" name="group_key" value="{{ group_key }}">
-    </paper-input>
-    <paper-button raised onclick="submitByFormId('users-group-form')" class="form-submit-button" type="submit">Fetch Users From Group</paper-button>
-  </form><br>
+  <paper-tabs selected="{{ selected }}">
+    <paper-tab>Add By Group</paper-tab>
+    <paper-tab>Add Individual User</paper-tab>
+    <paper-tab>Add By Domain</paper-tab>
+    <paper-tab>Add Manually</paper-tab>
+  </paper-tabs>
+
+  <iron-pages selected="{{ selected }}">
+    <div>
+      <form id="users-group-form" method="get"
+        action="{{ BASE_URL }}{{ user_add_path }}">
+        <paper-input label="Input a group key (group email address or unique id) to fetch more users."
+        type="textbox" name="group_key" value="{{ group_key }}">
+        </paper-input>
+        <br>
+        <paper-button raised onclick="submitByFormId('users-group-form')"
+        class="form-submit-button" type="submit">
+          Fetch Users From Group</paper-button>
+      </form>
+    </div>
+    <div>
+      <form id="users-user-form" method="get"
+        action="{{ BASE_URL }}{{ user_add_path }}">
+        <paper-input label="Input user key (email address or unique id) to search for a specific user."
+        type="email" name="user_key" value="{{ user_key }}">
+        </paper-input>
+        <br>
+        <paper-button raised onclick="submitByFormId('users-user-form')"
+        class="form-submit-button" type="submit">
+          Search for Specific User</paper-button>
+      </form>
+    </div>
+    <div>
+      <form id="users-domain-form" method="get"
+        action="{{ BASE_URL }}{{ user_add_path }}">
+        <input type="hidden" name="get_all" value="true">
+        <br>
+        <paper-button raised onclick="submitByFormId('users-domain-form')"
+        class="form-submit-button" type="submit">
+          Fetch All Users in Domain</paper-button>
+      </form>
+    </div>
+    <div>
+      <form id="users-manual-form" method="post"
+        action="{{ BASE_URL }}{{ user_add_path }}">
+        <paper-input label="Input user name here." type="textbox"
+        name="user_name" value="{{ user_name }}" required>
+        </paper-input>
+        <paper-input label="Input user email here." type="email"
+        name="user_email" value="{{ user_email }}" required>
```

Comment:
Also, should we be concerned if someone tries to add an email that doesn't exist?  Like 'john@foobar.com' instead of 'jonathan@foobar.com'?